content/docs/components/heading/theming.mdx

```diff
@@ -2,3 +2,160 @@
 id: heading
 scope: theming
 ---
+
+The `Heading` component is a single part component. All of the styling is applied
+directly to the `h2` element.
+
+> To learn more about styling single part components, visit the [Component Style](/docs/styled-system/component-style#styling-single-part-components) page
+## Theming properties
+
+The properties that affect the theming of the `Heading` component are:
+
+- `variant`: The visual variant of the divider.
+- `colorScheme`: The color scheme of the divider.
+- `size`: The size of the divider. Defaults to `xl`.
+
+## Theming utilities
+
+- `defineStyle`: a function used to create style objects.
+- `defineStyleConfig`: a function used to define the style configuration for a single part component.
+
+```js
+import { defineStyle, defineStyleConfig } from '@chakra-ui/react'
+```
+
+## Customizing the default theme
+
+```js
+import { defineStyle, defineStyleConfig } from '@chakra-ui/react'
+
+const custom = defineStyle({
+    color: "yellow.500",
+    border: "0px",
+    // let's also provide dark mode alternatives
+    _dark: {
+        color: 'yellow.300',
+    }
+})
+```
+After customizing the divider theme, we can import it in our theme file and add it in the components property:
+
+```js
+import { extendTheme } from '@chakra-ui/react'
+import { headingTheme } from './components/divider.ts'
+
+export const theme = extendTheme({
+    components: { Heading: headingTheme },
+})
+```
+
+> This is a crucial step to make sure that any changes that we make to the divider theme are applied.
+
+Every time you're adding anything new to the theme, you'd need to run the cli command to get proper autocomplete in your IDE. Here's how you can do that:
+
+```bash
+# run it once:
+npx chakra-cli tokens path/to/theme.ts
+# or run it in watch mode:
+npx chakra-cli tokens path/to/theme.ts --watch
+```
+
+## Adding a custom variant
+
+Let's assume we want to include a custom branded variant. Here's how we can do that:
+
+```js
+import { defineStyle, defineStyleConfig } from '@chakra-ui/react'
+
+const thicc = defineStyle({
+    color: "orange.500",
+    border: "0px",
+    bg: "blue.200",
+    p: "3",
+    borderRadius: "10",
+    transition: 'transform 0.15s ease-in-out, background 0.15s ease-in-out',
+    // let's also provide dark mode alternatives
+    _dark: {
+        color: 'orange.400',
+        bg: "blue.200"
+    },
+    _hover: {
+        bg: "blue.100",
+        color: "purple.300",
+        transform: "scale(1, 1)",
+        _dark: {
+            bg: "blue.100",
+            color: "purple.300",
+            transform: "scale(3, 3)",
+        }
+    }
+})
```

Comment:
This styling doesn't make sense for the `Heading` component. The `Heading` is a simple component. We can just change its `fontFamily`, `fontWeight`, and potentially adding a border at the bottom of it (underline). Also, rename `thicc` to something more suitable, like `underline` if you're adding a bottom border.